Src/ILGPU.Algorithms.Tests/VectorTests.cs

```diff
@@ -0,0 +1,412 @@
+﻿using ILGPU.Runtime;
+using ILGPU.Tests;
+using System.Numerics;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace ILGPU.Algorithms.Tests
+{
+    public interface IVector<T> : IXunitSerializable
+        where T : struct
+    {
+        T GetVector();
+    }
+    
+    public abstract partial class VectorTests : TestBase
+    {
+        protected VectorTests(ITestOutputHelper output, TestContext testContext)
+            : base(output, testContext)
+        { }
+
+        #region Xunit Vectors
+
+        #region Vector2
+
+        internal readonly struct Vector2Zero : IVector<Vector2>
+        {
+            Vector2 IVector<Vector2>.GetVector() => Vector2.Zero;
```

Comment:
question: is there any particular reason why we have to use an explicit interface implementation here?